### ISSUE TITLE
Type error: Argument 1 passed to PrestaShop\PrestaShop\Adapter\PDF\PDFGenerator::__construct() must be an instance of Smarty

### ISSUE THREAD

**Issue by bcsirt** [PrestaShop/PrestaShop#25077]
#### Describe the bug

After manual upgrade from PS 1.6 to 1.7.6.4, when trying to view dashboard page "Orders" => "Invoices" an Error 500 is thrown.

Enabling debug mode provides the following:

```
FatalThrowableError

Symfony\Component\Debug\Exception\FatalThrowableError:
Type error: Argument 1 passed to PrestaShop\PrestaShop\Adapter\PDF\PDFGenerator::__construct() must be an instance of Smarty, instance of SmartyDebug given, called in /var/www/vhosts/xxxxx.xx/httpdocs/var/cache/dev/ContainerExdmrlx/getPrestashop_Adapter_Pdf_Generator_InvoiceService.php on line 8

  at src/Adapter/PDF/PDFGenerator.php:53
  at PrestaShop\PrestaShop\Adapter\PDF\PDFGenerator->__construct(object(SmartyDebug), object(InvoiceTemplateTypeProvider))
     (var/cache/dev/ContainerExdmrlx/getPrestashop_Adapter_Pdf_Generator_InvoiceService.php:8)
  at require('/var/www/vhosts/xxxxx.xx/httpdocs/var/cache/dev/ContainerExdmrlx/getPrestashop_Adapter_Pdf_Generator_InvoiceService.php')
     (var/cache/dev/ContainerExdmrlx/appDevDebugProjectContainer.php:1919)
  at ContainerExdmrlx\appDevDebugProjectContainer->load('getPrestashop_Adapter_Pdf_Generator_InvoiceService.php')
     (var/cache/dev/ContainerExdmrlx/getPrestashop_Admin_Order_Invoices_ByDate_FormHandlerService.php:8)
  at require('/var/www/vhosts/xxxxx.xx/httpdocs/var/cache/dev/ContainerExdmrlx/getPrestashop_Admin_Order_Invoices_ByDate_FormHandlerService.php')
     (var/cache/dev/ContainerExdmrlx/appDevDebugProjectContainer.php:1919)
  at ContainerExdmrlx\appDevDebugProjectContainer->load('getPrestashop_Admin_Order_Invoices_ByDate_FormHandlerService.php')
     (vendor/symfony/symfony/src/Symfony/Component/DependencyInjection/Container.php:304)
  at Symfony\Component\DependencyInjection\Container->get('prestashop.admin.order.invoices.by_date.form_handler')
     (vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php:67)
  at Symfony\Bundle\FrameworkBundle\Controller\Controller->get('prestashop.admin.order.invoices.by_date.form_handler')
     (src/PrestaShopBundle/Controller/Admin/Sell/Order/InvoicesController.php:55)
  at PrestaShopBundle\Controller\Admin\Sell\Order\InvoicesController->indexAction(object(Request))
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:151)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/HttpKernel.php:68)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, false)
     (vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php:200)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request), 1, false)
     (backoffice/index.php:86)

```

#### Expected behavior

Explain what you expected to happen instead.

#### Steps to Reproduce

1. Go to 'Orders'
2. Click on 'Invoices'

#### Additional information

* PrestaShop version: 1.7.6.4
* PHP version: 7.1.33 
![2021-06-22_10-56-24](https://user-images.githubusercontent.com/4466339/122917297-2c62ef80-d35e-11eb-92e2-d88256f11f00.png)

**Comment by florine2623** [comment]
Hello @bcsirt ,

Could you clear cache and check again ?

Thanks!

**Comment by bcsirt** [comment]
> 
> 
> Hello @bcsirt ,
> 
> Could you clear cache and check again ?
> 
> Thanks!

Yes, but no effect.

**Comment by florine2623** [comment]
If you cleared cache from your files, the error should be a different one, can you send it back ?

**Comment by bcsirt** [comment]
Hello, I tried to use autoupgrade to 1.7.6.9 version but I have the same issue with invoices tab.
When I cleared cache, I had the same error.  I installed my store locally, then I deleted all the orders, customers with the pscleaner module and I have the same problem with the invoices tab.

Has anyone encountered this bug before?

Regards,

**Comment by bcsirt** [comment]
![2021-06-25_12-08-22](https://user-images.githubusercontent.com/4466339/123409291-4e52b100-d5ae-11eb-96e1-7ba78b43de12.png)

**Comment by khouloudbelguith** [comment]
Hi @bcsirt,

Thanks for your feedback.
I have an exception with PS1769 only (not reproduced with PS1775) following these steps:
1. Create a product with text **customization** required
2. Create an order with this Product
3. Generate an invoice for this Order
4. Go to BO > Orders > Invoices > Generate invoice by date or by status
5. See exception 
![image](https://user-images.githubusercontent.com/16067358/123415040-f8cdd280-d5b4-11eb-8fa0-aed3edbf980c.png)

I tried with PS1775 => ok
 
Could you please confirm for me that this issue occurs only with a customized product and could you check with the last version PS1775?

https://drive.google.com/file/d/1RgORqGdGQugttjsgF_xvKTjRbCJQazJ5/view?usp=sharing

About the pscleaner module: the module has been archived, it’s no longer maintained.

Thank you!

**Comment by bcsirt** [comment]
Re, same issue with 1.7.7.5 version :( 
![2021-06-25_13-24-12](https://user-images.githubusercontent.com/4466339/123418015-d2119b00-d5b8-11eb-81e1-b9b8bac1c109.png)

**Comment by khouloudbelguith** [comment]
@bcsirt, did you make an upgrade to PS1.7.7.5 or it is a fresh install PS1.7.7.5?
This issue occurs only when you delete orders and customers with the pscleaner module or you have this issue following my steps added in my previous comment?

Thank you!

**Comment by bcsirt** [comment]
On a blank installation there are no problems on versions 1.7.6.4 , 1.7.6.9 and 1.7.7.5. But I want to fix the error on the store without having to reinstall everything.
During my tests I emptied the contents of the tables to be sure that there was no corrupted data. This said, the problem with the invoice tab is present in the tables orders, customers full

**Comment by bcsirt** [comment]
Yes, I have the same issue with your process.

**Comment by khouloudbelguith** [comment]
Hi @bcsirt,

I made an upgrade from PS16124 to PS1775 using the 1-click upgrade v4.12.0.
https://drive.google.com/file/d/1RMXhKPOp7VbIHLLSP3Zy0rT5CRwCPvK3/view?usp=sharing

I made an upgrade from PS16124 to PS1769 using the 1-click upgrade v4.12.0.
https://drive.google.com/file/d/1WfUT7ltId8VbBcak2t4clZhBqWT9amUD/view?usp=sharing

After an upgrade, I tried to generate an invoice => But no invoice generated: this is NOK but I don't have any exception
Thanks to check and feedback.
